lib/cn/csched_sp3.c

```diff
@@ -2493,12 +2500,16 @@ sp3_monitor(struct work_struct *work)
             bad_health = true;
         }
 
-        if (sp->activity) {
-            last_activity = now + NSEC_PER_SEC * 5;
-            sp->activity = 0;
+        if (now > chk_sched.next || sp->activity) {
+            if (sp->activity) {
+                last_activity = now + NSEC_PER_SEC * 5;
+                sp->activity = 0;
+            }
 
             if (!bad_health)
                 sp3_schedule(sp);
+
+            chk_sched.next = now + chk_sched.interval;
```

Comment:
The bug was that we were not calling sp3_schedule() unless there was activity (e.g., an ingest), which meant that timer-based work would not be scheduled if the tree was otherwise idle.
Now we call sp3_schedule() at least three times per second if there is activity, and at least once every three seconds otherwise...